### ISSUE TITLE
Calculating the RMSE

### ISSUE THREAD

**Issue by ms-program** [ReynoldsJ20/SlicerFiducialsToModelDistance#7]
I want to use the Fiducials To Model Distance modelu to calculate the RMSE. I know that RMSE is a point-to-point measure, but if I specify a segment in the “Input reference points or surface” field as shown in the figure, what will the RMSE be based on in the “Input reference points” and the segment?
I would appreciate your answer.
![image](https://user-images.githubusercontent.com/68321071/121299290-26b5e480-c930-11eb-8b94-1e752a4d73ca.png)

**Comment by ReynoldsJ20** [comment]
The module does not allow a segment to be input into the "Input reference points or surface" area. The only valid inputs here are either fiducial points or a surface model. The output point distances table shows the shortest distance between each fiducial point and the surface of the model. The RMS distance is calculated using the formula in the image below.

![image](https://user-images.githubusercontent.com/60028999/121304581-a5227e80-c950-11eb-99a2-27671174918d.png)

Say there are three fiducial points as shown in the image below:

![image](https://user-images.githubusercontent.com/60028999/121305692-0008a580-c952-11eb-9bad-e5be42e566a1.png)

and the output point distances table gives the following point to surface distances:

![image](https://user-images.githubusercontent.com/60028999/121304875-03e7f800-c951-11eb-9ef4-526469fcb0da.png)

The RMS distance is calculated as sqrt( (0.2479^2 + 1.28345^2 + 1.6572^2) / 3) = 1.21868 as show in the error metrics table:

![image](https://user-images.githubusercontent.com/60028999/121305142-5a553680-c951-11eb-9639-e13d6de13e5f.png)

Hope that helps? Also you should use different tables for the "output error metrics table" and "output point distances table" I think the program is going to overwrite the "Error metrics" table as you are using this for both inputs.

**Comment by ms-program** [comment]
Thanks for the answer.
Do you mean that I should convert the segment to a model and specify it as "Input reference points or surface" instead of leaving it as a segment?(fig1)
(fig1)
![image](https://user-images.githubusercontent.com/68321071/121309145-cda07d80-c93c-11eb-9c6b-f0b5e3ca5670.png)

Is it correct that the way to use this module is to specify the "Input point" based on CT images, etc., and then calculate the RMSE based on the points in the "Input reference points or surface" or the model surface?
Also, if you use the model surface, where on the model surface do you compare the "Input point"?
I apologize for the inconvenience, but I would appreciate your answer.

**Comment by ReynoldsJ20** [comment]
Yes you need to convert the segments into a model. The input points has to be a set of fiducial points which are just points in 3D space. As shown in the image of the green skull above. There are three fiducial points shown: F-1, F-2 and F-3.

**Comment by ms-program** [comment]
Thank you for your answer.
I would like to ask an additional question.
1. "The input points have to be a set of fiducial points which are just points in 3D space. As shown in the image of the green skull above."
→Should I use this tool to create the points?
![image](https://user-images.githubusercontent.com/68321071/121331620-026aff80-c952-11eb-8bb2-9aa83d084260.png)

2.When I select the model surface instead of points in "Input reference points or surface", what does this module refer to on the model surface to give me each value?
I would appreciate it if you could tell me.

**Comment by ReynoldsJ20** [comment]
1) Yes that is the button to make fiducial points

2) The distances shown on the "output distances table" are the shortest possible distance between the fiducial point and the surface of the model. I have made an example in the image below. I have placed the point F_2 some distance away from the surface of the model. You can see from the ruler that I have measured approximately the shortest distance between the fiducial point and the surface of the model. In the 3d view the measurement is 8.98 which is very close to the Distance value in the table for that same point (8.96763 for F_2). The value in the table is more accurate because I just quickly put the ruler there to demonstrate what it is measuring.

![image](https://user-images.githubusercontent.com/60028999/121336560-c0e94d00-c96f-11eb-834a-596c9e157ee7.png)

**Comment by ms-program** [comment]
Thank you for your prompt reply .
Sorry for asking so many times.
The markup point can only be hit on one slice surface, but the model is stored in series on all slices. Does it take into account the slice where the point is located and the model to calculate the minimum value for each distance and RMSE?

**Comment by ReynoldsJ20** [comment]
The distance is the minimum distance in 3d space. So yes it takes into account that the model is 3d and not just a single slice.

**Comment by ms-program** [comment]
Thank you for your prompt reply .
If I have any more questions, I would like to ask you.
Thank you very much.
